components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,541 @@
+	local store = category or _CLEAN_OTHER_ROLES[role] or 'Player'
+
+	return { title = 'Race', display = raceData.display, store = store, category = category or 'Player'}
+end
+
+function StarCraft2Player:calculateEarnings(args)
+	shouldStoreData = Player.shouldStoreData(args)
+
+	if shouldStoreData then
+		local earningsTotal
+		earningsTotal, earningsGlobal = StarCraft2Player._getEarningsMedalsData(self.pagename)
+		earningsTotal = math.floor( (earningsTotal or 0) * 100 + 0.5) / 100
+		return earningsTotal
+	end
+	return 0
+end
+
+function StarCraft2Player._getLPDBrecursive(cond, query, queryType)
+	local data = {} -- get LPDB results in here
+	local count
+	local offset = 0
+	repeat
+		local additionalData = mw.ext.LiquipediaDB.lpdb(queryType, {
+			conditions = cond,
+			query = query,
+			offset = offset,
+			limit = 5000
+		})
+		count = #additionalData
+		-- Merging
+		for i, item in ipairs(additionalData) do
+			data[offset + i] = item
+		end
+		offset = offset + count
+	until count ~= 5000
+
+	return data
+end
+
+function StarCraft2Player._getEarningsMedalsData(player)
+	local cond = '[[date::!1970-01-01 00:00:00]] AND ([[prizemoney::>0]] OR ' ..
+		'([[mode::1v1]] AND ([[placement::1]] OR [[placement::2]] OR [[placement::3]]' ..
+		' OR [[placement::4]] OR [[placement::3-4]]))) AND ' ..
+		'[[participant::' .. player .. ']]'
+	local query = 'liquipediatier, liquipediatiertype, placement, date, prizemoney, mode'
+
+	local data = StarCraft2Player._getLPDBrecursive(cond, query, 'placement')
+
+	local earnings = {}
+	local Medals = {}
+	earnings['total'] = {}
+	Medals['total'] = {}
+	local earnings_total = 0
+
+	if type(data[1]) == 'table' then
+		for i = 1, #data do
+			--handle earnings
+			local mode = _EARNING_MODES[data[i].mode] or 'other'
+			if not earnings[mode] then
+				earnings[mode] = {}
+			end
+			local date = string.sub(data[i].date, 1, 4)
+			if not earnings[mode][date] then
+				earnings[mode][date] = 0
+			end
+			if not earnings['total'][date] then
+				earnings['total'][date] = 0
+			end
+			earnings[mode][date] = earnings[mode][date] + data[i].prizemoney
+			earnings['total'][date] = earnings['total'][date] + data[i].prizemoney
+			earnings_total = earnings_total + data[i].prizemoney
+
+			--handle medals
+			if data[i].mode == '1v1' then
+				local place = StarCraft2Player._Placements(data[i].placement)
+				if place ~= _DISCARD_PLACEMENT then
+					local tier = data[i].liquipediatier or 'undefined'
+					if data[i].liquipediatiertype ~= 'Qualifier' then
+						if not Medals[place] then
+							Medals[place] = {}
+						end
+						if not Medals[place][tier] then
+							Medals[place][tier] = 0
+						end
+						if not Medals[place]['total'] then
+							Medals[place]['total'] = 0
+						end
+						Medals[place][data[i].liquipediatier or ''] = Medals[place][tier] + 1
+						Medals[place]['total'] = Medals[place]['total'] + 1
+						if not Medals['total'][tier] then
+							Medals['total'][tier] = 0
+						end
```

Comment:
`if not Medals['total'][tier] then
	Medals['total'][tier] = 0
end`
is done twice. Also see other comments for suggestion to remove it entirely